test/utilities/testsUtils.ts

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { TreeItem, Workbench, SideBarView, ViewSection } from 'wdio-vscode-service';
+import { Duration, log, pause } from './miscellaneous.ts';
+import { attemptToFindOutputPanelText } from './outputView.ts';
+import { notificationIsPresentWithTimeout } from './notifications.ts';
+import { getTerminalViewText } from './terminalView.ts';
+import { fail } from 'assert';
+
+const CONTINUE = 'F5';
+
+export async function retrieveExpectedNumTestsFromSidebar(
+  expectedNumTests: number,
+  testsSection: ViewSection,
+  actionLabel: string
+): Promise<TreeItem[]> {
+  let testsItems = (await testsSection.getVisibleItems()) as TreeItem[];
+  await browser.keys(['Escape']);
+
+  // If the tests did not show up, click the refresh button on the top right corner of the Test sidebar
+  for (let x = 0; x < 3; x++) {
+    if (testsItems.length === 1) {
+      await testsSection.elem.click();
+      const refreshAction = await testsSection.getAction(actionLabel);
+      if (!refreshAction) {
+        fail('Could not find refresh tests action button');
+      }
+      await refreshAction.elem.click();
+      await pause(Duration.seconds(10));
+      testsItems = (await testsSection.getVisibleItems()) as TreeItem[];
+    } else if (testsItems.length === expectedNumTests) {
+      break;
+    }
+  }
+
+  return testsItems;
+}
+
+export async function getTestsSection(workbench: Workbench, type: string) {
+  const sidebar = workbench.getSideBar();
+  const sidebarView = sidebar.getContent();
+  const testsSection = await sidebarView.getSection(type);
+  await expect(testsSection.elem).toBeDefined();
+  return testsSection;
+}
+
+/**
+ * Runs a test case from the sidebar and returns the test result.
+ * *
+ * @param {Workbench} workbench - The workbench instance used to interact with the sidebar and views.
+ * @param {string} testSuite - The name of the test suite from which to run the test (e.g., 'APEX TESTS', 'LWC TESTS').
+ * @param {string} testName - The name of the specific test case to run.
+ * @param {string} actionLabel - The label of the action button to click (e.g., 'SFDX: Run Apex Tests', 'Debug').
```

Comment:
```suggestion
 * @param {string} actionLabel - The label of the action button to click (e.g., 'Run Single Test', 'SFDX: Run Lightning Web Component Test File').
```

This function is used only for run, not debug.  Also, the buttons for Apex are "Run Tests" and "Run Single Test".